src/vs/workbench/services/remote/common/remoteEnvironmentService.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IRemoteAgentService } from 'vs/workbench/services/remote/common/remoteAgentService';
+import { IRemoteAgentEnvironment } from 'vs/platform/remote/common/remoteAgentEnvironment';
+import { registerSingleton } from 'vs/platform/instantiation/common/extensions';
+import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+
+export interface IRemoteEnvironmentService {
+	_serviceBrand: any;
+	remoteEnvironment: Promise<IRemoteAgentEnvironment | undefined>;
```

Comment:
Small nitpick: Please use `IRemoteAgentEnvironment | null` for consistency with `IRemoteAgentConnection.getEnvironment` which also returns `Promise<IRemoteAgentEnvironment | null>`